app/webpacker/components/ContactEditProfilePage/EditProfileForm.jsx

```diff
@@ -0,0 +1,185 @@
+import { apiV0Urls, contactEditProfileActionUrl } from '../../lib/requests/routes.js.erb';
+import { genders, countries } from '../../lib/wca-data.js.erb';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+import useSaveAction from '../../lib/hooks/useSaveAction';
+import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
+import UtcDatePicker from '../wca/UtcDatePicker';
+
+const CONTACT_EDIT_PROFILE_FORM_QUERY_CLIENT = new QueryClient();
+
+const genderOptions = _.map(genders.byId, (gender) => ({
+  key: gender.id,
+  text: gender.name,
+  value: gender.id,
+}));
+
+const countryOptions = _.map(countries.byIso2, (country) => ({
+  key: country.iso2,
+  text: country.name,
+  value: country.iso2,
+}));
+
+export default function EditProfileForm({
+  wcaId: loggedInUserWcaId,
+  loggedInUserData,
+  onContactSuccess,
+  recaptchaPublicKey,
+}) {
+  const [contactEmail, setContactEmail] = useState();
+  const [customWcaId, setCustomWcaId] = useState();
+  const [editProfileReason, setEditProfileReason] = useState();
+  const [editedProfileDetails, setEditedProfileDetails] = useState();
+  const [proof, setProof] = useState();
+  const [captchaValue, setCaptchaValue] = useState();
+  const [captchaError, setCaptchaError] = useState(false);
+  const [saveError, setSaveError] = useState();
+  const { save, saving } = useSaveAction();
+  const wcaId = loggedInUserWcaId || customWcaId;
+  const { data, isLoading, isError } = useQuery({
+    queryKey: ['profileData'],
+    queryFn: () => fetchJsonOrError(apiV0Urls.persons.show(wcaId)),
+  }, CONTACT_EDIT_PROFILE_FORM_QUERY_CLIENT);
+  const profileDetails = data?.data?.person;
+
+  useEffect(() => {
+    setEditedProfileDetails(profileDetails);
+  }, [profileDetails]);
+
+  const formSubmitHandler = () => {
+    const formData = new FormData();
+    formData.append('formValues', JSON.stringify({
+      editedProfileDetails, editProfileReason, wcaId, contactEmail,
+    }));
+    formData.append('attachment', proof);
+    save(
+      contactEditProfileActionUrl,
+      formData,
+      onContactSuccess,
+      { method: 'POST', headers: {}, body: formData },
+      setSaveError,
+    );
+  };
+
+  const handleEditProfileReaconChange = (e, { value }) => {
+    setEditProfileReason(value);
+  };
+
+  const handleProofUpload = (event) => {
+    setProof(event.target.files[0]);
+  };
+
+  const handleFormChange = (e, { name: formName, value }) => {
+    setEditedProfileDetails((prev) => ({ ...prev, [formName]: value }));
+  };
+
+  if (saving || isLoading) return <Loading />;
+  if (saveError || isError) return <Errored />;
+
+  return (
+    <Form onSubmit={formSubmitHandler}>
+      {!loggedInUserData && (
+        <Form.Input
+          label={i18n.t('page.contact_edit_profile.form.contact_email.label')}
+          value={contactEmail}
+          onChange={(e, { value }) => setContactEmail(value)}
```

Comment:
If you declare `contactEmail` with our WCA custom `useInputState` instead of `useState`, you can pass the `setContactEmail` setter here directly and it will "just work" :magic_wand: